### PR TITLE
feat(artifacts) Implement S3 artifacts

### PR DESCRIPTION
**Pull request by benjaminws**
Implement S3 artifacts in the UI to support https://github.com/spinnaker/spinnaker/issues/2595

Looking for feedback.

### CODE REVIEW

File: app/scripts/modules/core/src/pipeline/config/triggers/artifacts/s3/defaultS3.artifact.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        }
+      };
+    },
+    controllerAs: 'ctrl',
+    template: `
+<div class="col-md-12">
+  <div class="form-group row">
+    <label class="col-md-2 sm-label-right">
+      Object path
+      <help-field key="pipeline.config.expectedArtifact.defaultS3.reference"></help-field>
```
Comment (sbwsg) [inline app/scripts/modules/core/src/pipeline/config/triggers/artifacts/s3/defaultS3.artifact.ts]: this key and the one in s3.artifact.ts need correlating help text entries in `app/scripts/modules/core/src/help/help.contents.ts`

Comment (benjaminws) [reply app/scripts/modules/core/src/pipeline/config/triggers/artifacts/s3/defaultS3.artifact.ts]: Awesome, I'll work that out. Should have figured that was referencing _something_ ;)

File: app/scripts/modules/core/src/pipeline/config/triggers/artifacts/s3/defaultS3.artifact.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 Netflix, Inc.
```
Comment (danielpeach) [inline app/scripts/modules/core/src/pipeline/config/triggers/artifacts/s3/defaultS3.artifact.ts]: Omit this - not needed for Deck.

File: app/scripts/modules/core/src/pipeline/config/triggers/artifacts/s3/s3.artifact.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 Netflix, Inc.
```
Comment (danielpeach) [inline app/scripts/modules/core/src/pipeline/config/triggers/artifacts/s3/s3.artifact.ts]: Here too.

**Comment by benjaminws** [comment]
Looks like my editor automatically added this copy stuff, probably because
I used gradle helpers to set up the editor, will wipe it out.

File: app/scripts/modules/core/src/pipeline/config/triggers/artifacts/s3/s3.artifact.ts (hunk 3 above)
Comment (benjaminws) [reply app/scripts/modules/core/src/pipeline/config/triggers/artifacts/s3/s3.artifact.ts]: done

**Comment by lwander** [comment]
Any objections to merging @sbwsg or @danielpeach ?

**Pull request merged**